### DISCUSSION TITLE
OpenSSL 3.0.8 FIPS throwing Provider routines::missing config data error

### DISCUSSION THREAD

**Discussion by danThertell** [openssl/openssl#23100] [Q&A]
Hey,

So I am trying to build openssl 3.0.8 on RHEL8 with the fips module enabled.
These are the steps i have taken to try and make it work.

1. Download openssl3.0.8 source as tar
2. unpack tar
3. run "./Configure enable-fips" I have also tried adding --openssldir and --prefix and pointing them to non standard paths, but the error remains the same
4. run "make"
5. run "sudo make install"
6. run "openssl fipsinstall -module ${pathToFIPS.so} -out ${pathToFipsmodule.cnf} -provider_name fips"
7. verify the install using "openssl fipsinstall -module ${pathToFIPS.so} -in ${pathToFipsmodule.cnf} -provider_name fips verify" and I receive "VERIFY PASSED"
8. update openssl.cnf to include the full path to fipsmodule.cnf
9. update opnessl.cnf, uncomment "#fips = fips_sect"
10. compile and run my program that wraps openssl, and call OSSL_LIB_CTX_load_config() with the full path to the modified openssl.cnf listed above. This step does not fail.
11. call OSSL_PROVIDER_load(libPtr, "fips"), this step fails, and the top error in the error queue is "Provider routines::missing config data error"

Has anyone had this issue before? Does anyone know what i might be doing wrong to cause this issue? I have tried to implement the fixes and changes mentioned in a few other threads without any luck.  If you need any other details about the setup please let me know and I will post those details.

Thanks,
Dan

**Comment by danThertell** [comment]
I have tried debugging it with GDB a few times.
GDB shows the error being thrown on line 386 in self_text.c
I am not sure why st->indicator_checksum_data is null

**Comment by t8m** [comment]
I'd suggest using the `strace` command to see whether the openssl.cnf and fipsmodule.cnf is being properly loaded. Can you also paste the contents of openssl.cnf and fipsmodule.cnf here?

One thing you could also try is to set the OPENSSL_CONF environment variable to point at the openssl.cnf file.

**Comment by mattcaswell** [comment]
You might also be interested in my recent talk on setting up and configuring the FIPS provider:

https://www.youtube.com/watch?v=geAtEXbHaFg
